enhancements/cluster-capabilities-api.md

```diff
@@ -0,0 +1,328 @@
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to make use of the enhancement?
+
+Upgrade expectations:
+- Each component should remain available for user requests and
+  workloads during upgrades. Any exception to this should be
+  identified and discussed here.
+- Micro version upgrades - users should be able to skip forward versions within a
+  minor release stream without being required to pass through intermediate
+  versions - i.e. `x.y.N->x.y.N+2` should work without requiring `x.y.N->x.y.N+1`
+  as an intermediate step.
+- Minor version upgrades - you only need to support `x.N->x.N+1` upgrade
+  steps. So, for example, it is acceptable to require a user running 4.3 to
+  upgrade to 4.5 with a `4.3->4.4` step followed by a `4.4->4.5` step.
+- While an upgrade is in progress, new component versions should
+  continue to operate correctly in concert with older component
+  versions (aka "version skew"). For example, if a node is down, and
+  an operator is rolling out a daemonset, the old and new daemonset
+  pods must continue to work correctly even while the cluster remains
+  in this partially upgraded state for some time.
+
+Downgrade expectations:
+- If an `N->N+1` upgrade fails mid-way through, or if the `N+1` cluster is
+  misbehaving, it should be possible for the user to rollback to `N`. It is
+  acceptable to require some documented manual steps in order to fully restore
+  the downgraded cluster to its previous state. Examples of acceptable steps
+  include:
+  - Deleting any CVO-managed resources added by the new version. The
+    CVO does not currently delete resources that no longer exist in
+    the target version.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should _not_ be implemented.
+
+## Alternatives
+
+### Add fields to existing configuration APIs
+
+We could add the fields we know about today to the
+`infrastructure.config.openshift.io` API, instead of creating a new
+API. This would be less work, initially, but muddies the meaning of
+the APIs.
```

Comment:
Current `Infrastructure` scope [is][1]:

> Infrastructure holds cluster-wide information about Infrastructure.

Which is already fairly muddy ;).  It's not clear to me that HA-ness or target resource consumption is less infrastructure-y than [API URIs][2], and those are already part of the `Infrastructure` spec.  I think the trade-off here is between recycling an existing CRD (less dev and doc work, possibly slight efficiency savings in API-server memory or something?  I don't have a clear understanding of per-CRD resource use) and creating a new one (ideally clearer scoping, if we're worried about overloading an existing CRD).

[1]: https://github.com/openshift/api/blob/c6debb38648fa2094d36ef7dd2ffc0dca97da879/config/v1/types_infrastructure.go#L10
[2]: https://github.com/openshift/api/blob/c6debb38648fa2094d36ef7dd2ffc0dca97da879/config/v1/types_infrastructure.go#L70-L79

Reply:
Yeah, I don't have a particularly strong opinion on this. It would be less work in the short term to just extend the existing resource, but it seemed cleaner to add a new one. I'll go along with whatever the API owners agree to do.